Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Document/ShaderManagementConsoleDocument.cpp

```diff
@@ -197,109 +266,106 @@ namespace ShaderManagementConsole
         AZStd::string shaderPath = m_shaderVariantListSourceData.m_shaderFilePath;
 
         auto shaderAssetResult = AZ::RPI::AssetUtils::LoadAsset<AZ::RPI::ShaderAsset>(m_absolutePath, shaderPath);
-        if (shaderAssetResult)
+        if (!shaderAssetResult)
         {
-            m_shaderAsset = shaderAssetResult.GetValue();
+            AZ_Error("ShaderManagementConsoleDocument", false, "Could not load shader asset: %s.", shaderPath.c_str());
+            return;
+        }
+        m_shaderAsset = shaderAssetResult.GetValue();
+
+        // We consider an empty shader variant list data set, a request for initialization
+        if (m_shaderVariantListSourceData.m_shaderVariants.empty())
```

Comment:
Should we make this assumption? This function is called from the table view and other places to update data. If someone is intentionally setting uninitialized, empty, or cleared shader variant source data should that be a request for initialization? It seems like that should be a separate function, called explicitly.

Reply:
In this PR, that's only a change in indentation. That code was there before, though I did change the comment in a previous PR; in hope to trigger a discussion on that question. My analysis is that the condition is inappropriate because of  what you say. I didn't zealously change the behavior because it appears it's only a tiny performance pessimization but not an "invariants hazard" (i.e. it is still functional).